indico/modules/events/registration/badges.py

```diff
@@ -107,4 +108,75 @@ def _draw_badge(self, canvas, registration, pos_x, pos_y):
             elif item['text']:
                 text = item['text']
 
-            self._draw_item(canvas, item, text, pos_x, pos_y)
+            self._draw_item(canvas, item, tpl_data, text, pos_x, pos_y)
+
+
+class RegistrantsListToBadgesPDFFoldable(RegistrantsListToBadgesPDF):
+    def _build_pdf(self, canvas):
+        # Only one badge per page
+        n_horizontal = 1
+        n_vertical = 1
+
+        for registration, (x, y) in izip(self.registrations, self._iter_position(canvas, n_horizontal, n_vertical)):
+            self._draw_badge(canvas, registration, self.template, self.tpl_data, x * cm, y * cm)
+            if self.tpl_data.width > self.tpl_data.height:
+                canvas.translate(self.width, self.height)
+                canvas.rotate(180)
+                self._draw_badge(canvas, registration, self.template.backside_template, self.backside_tpl_data,
+                                 self.width - self.tpl_data.width_cm * cm, y * cm)
+                canvas.translate(0, 0)
+                canvas.rotate(180)
+            else:
+                self._draw_badge(canvas, registration, self.template.backside_template, self.backside_tpl_data,
+                                 self.tpl_data.width_cm * cm, y * cm)
+
+
+class RegistrantsListToBadgesPDFDoubleSided(RegistrantsListToBadgesPDF):
+    def _build_pdf(self, canvas):
+        config = self.config
+
+        available_width = self.width - (config.left_margin - config.right_margin + config.margin_columns) * cm
+        n_horizontal = int(available_width / ((self.tpl_data.width_cm + config.margin_columns) * cm))
+        available_height = self.height - (config.top_margin - config.bottom_margin + config.margin_rows) * cm
+        n_vertical = int(available_height / ((self.tpl_data.height_cm + config.margin_rows) * cm))
+
+        if not (n_horizontal and n_vertical):
+            raise ValueError(_("The template dimensions are too large for the page size you selected"))
+
+        badges_mix = []
+        counter = 0
+        to_repeat = []
+        one_page = n_horizontal * n_vertical
+        # make batch of as many badges as we can fit into one page and add duplicates for printing back sides
+        for reg in self.registrations:
+            counter += 1
+            badges_mix.append(reg)
+            to_repeat.append(reg)
+            if counter == one_page:
```

Comment:
It looks a lot like you can use `enumerate()` when iterating over the registrations and use the module operator to check whether to run the code here (`(counter % one_page) == 0`)

I'd also rename `counter` to the more common `i` for such a variable.